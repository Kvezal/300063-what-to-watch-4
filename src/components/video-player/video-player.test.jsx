import React from "react";
import render from "react-test-renderer";
import Enzyme, {mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import VideoPlayer from "./video-player";


Enzyme.configure({
  adapter: new Adapter(),
});

describe(`VideoPlayerComponent`, () => {
  const source = `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`;
  const poster = `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`;

  test(`should render component`, () => {
    const tree = render
      .create(
          <VideoPlayer
            source={source}
            poster={poster}
            isPlaying={false}
            muted={true}
          />,
          {createNodeMock: () => ({})}
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test(`should create component`, () => {
    const videoPlayerComponent = mount(
        <VideoPlayer
          source={source}
          poster={poster}
          isPlaying={false}
          muted={true}
        />
    );
    const videoPlayer = videoPlayerComponent.find(`video`);
    expect(videoPlayer).toHaveLength(1);
  });
});
