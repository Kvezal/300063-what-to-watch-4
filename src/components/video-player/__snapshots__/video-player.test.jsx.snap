// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayerComponent should render component 1`] = `
<video
  width="100%"
/>
`;
